1.5.4 build failures

When forcing `transformers-0.6`:

```
                     9.2.0.20210422  9.0.1  8.10.4  8.8.4  8.6.5  8.4.4  8.2.2  8.0.2  7.10.3  7.8.4  7.6.3  7.4.2  7.2.2  7.0.4
contravariant-1.5.4  OK              OK     OK      OK     OK     FAIL   FAIL   FAIL   NO-IP   NO-IP  NO-IP  NO-IP  NO-IP  NO-IP
contravariant-1.5.3  NO-IP           NO-IP  NO-IP   NO-IP  NO-IP  NO-IP  NO-IP  NO-IP  NO-IP   NO-IP  NO-IP  NO-IP  NO-IP  NO-IP
contravariant-1.5.2  NO-IP           NO-IP  NO-IP   NO-IP  NO-IP  NO-IP  NO-IP  NO-IP  NO-IP   NO-IP  NO-IP  NO-IP  NO-IP  NO-IP
contravariant-1.5.1  NO-IP           NO-IP  NO-IP   NO-IP  NO-IP  NO-IP  NO-IP  NO-IP  NO-IP   NO-IP  NO-IP  NO-IP  NO-IP  NO-IP
contravariant-1.5    NO-IP           NO-IP  NO-IP   NO-IP  NO-IP  NO-IP  NO-IP  NO-IP  NO-IP   NO-IP  NO-IP  NO-IP  NO-IP  NO-IP

Build profile: -w ghc-8.4.4 -O0
In order, the following will be built (use -v for more details):
 - contravariant-1.5.4 (lib) (first run)
Configuring library for contravariant-1.5.4..
Preprocessing library for contravariant-1.5.4..
Building library for contravariant-1.5.4..
[1 of 4] Compiling Data.Functor.Contravariant ( old-src/Data/Functor/Contravariant.hs, /codetmp/con/contravariant-1.5.4/.dist-newstyle-trustee/13cc291791f26ca51aed2851dd003eba0257780261849c6556379a682286357e05ec60ce84c38ff6eed02c0d1dfc5ef4c981d84b74e29c70a8ffd4569974e253/build/x86_64-linux/ghc-8.4.4/contravariant-1.5.4/noopt/build/Data/Functor/Contravariant.o )


old-src/Data/Functor/Contravariant.hs:69:1: error:
    Could not find module ‘Control.Monad.Trans.Error’
    Perhaps you meant
      Control.Monad.Trans.Writer (from transformers-0.6.0.2)
      Control.Monad.Trans.Accum (from transformers-0.6.0.2)
      Control.Monad.Trans.Class (from transformers-0.6.0.2)
    Use -v to see a list of the files searched for.
   |
69 | import Control.Monad.Trans.Error
```

EDIT: I think relaxation of https://github.com/ekmett/semigroups/issues/101 would make older GHCs have (failing) installing plans with `contravariant-1.5.4` and `transformers-0.6`, so probably bette to decide what to do with `semigroups` first.

--- a/CHANGELOG.markdown
+++ b/CHANGELOG.markdown
@@ -1,3 +1,7 @@
+next [????.??.??]
+-----------------
+* Fix the build on old GHCs using `transformers-0.6.*`.
+
 1.5.4 [2021.07.25]
 ------------------
 * Allow building with `transformers-0.6.*`.

--- a/old-src/Data/Functor/Contravariant.hs
+++ b/old-src/Data/Functor/Contravariant.hs
@@ -20,7 +20,9 @@
 {-# LANGUAGE Trustworthy #-}
 #endif
 
+#if !(MIN_VERSION_transformers(0,6,0))
 {-# OPTIONS_GHC -fno-warn-deprecations #-}
+#endif
 
 -----------------------------------------------------------------------------
 -- |
@@ -66,10 +68,8 @@ import Control.Applicative.Backwards
 
 import Control.Category
 
-import Control.Monad.Trans.Error
 import Control.Monad.Trans.Except
 import Control.Monad.Trans.Identity
-import Control.Monad.Trans.List
 import Control.Monad.Trans.Maybe
 import qualified Control.Monad.Trans.RWS.Lazy as Lazy
 import qualified Control.Monad.Trans.RWS.Strict as Strict
@@ -86,6 +86,11 @@ import Data.Functor.Sum
 import Data.Functor.Constant
 import Data.Functor.Compose
 import Data.Functor.Reverse
+
+#if !(MIN_VERSION_transformers(0,6,0))
+import Control.Monad.Trans.Error
+import Control.Monad.Trans.List
+#endif
 
 #if MIN_VERSION_base(4,8,0)
 import Data.Monoid (Alt(..))
@@ -222,17 +227,11 @@ instance (Contravariant f, Contravariant g) => Contravariant (f :+: g) where
   contramap f (R1 ys) = R1 (contramap f ys)
 #endif
 
-instance Contravariant m => Contravariant (ErrorT e m) where
-  contramap f = ErrorT . contramap (fmap f) . runErrorT
-
 instance Contravariant m => Contravariant (ExceptT e m) where
   contramap f = ExceptT . contramap (fmap f) . runExceptT
 
 instance Contravariant f => Contravariant (IdentityT f) where
   contramap f = IdentityT . contramap f . runIdentityT
-
-instance Contravariant m => Contravariant (ListT m) where
-  contramap f = ListT . contramap (fmap f) . runListT
 
 instance Contravariant m => Contravariant (MaybeT m) where
   contramap f = MaybeT . contramap (fmap f) . runMaybeT
@@ -286,6 +285,14 @@ instance Contravariant f => Contravariant (Backwards f) where
 instance Contravariant f => Contravariant (Reverse f) where
   contramap f = Reverse . contramap f . getReverse
   {-# INLINE contramap #-}
+
+#if !(MIN_VERSION_transformers(0,6,0))
+instance Contravariant m => Contravariant (ErrorT e m) where
+  contramap f = ErrorT . contramap (fmap f) . runErrorT
+
+instance Contravariant m => Contravariant (ListT m) where
+  contramap f = ListT . contramap (fmap f) . runListT
+#endif
 
 #ifdef MIN_VERSION_StateVar
 instance Contravariant SettableStateVar where

